components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/util/IdPManagementConstants.java

```diff
@@ -604,6 +604,14 @@ public static class SQLQueries {
                 "SELECT * FROM IDP_AUTHENTICATOR WHERE TENANT_ID = ? AND DEFINED_BY = 'USER'";
     }
 
+    public static class WarningMessage {
+
+        public static final String WARN_STALE_IDP_ACTION = "The %s Identity Provider was deleted, but an error " +
+                "occurred while deleting its associated action.";
+        public static final String WARN_STALE_IDP_ACTIONS = "The Identity Providers was deleted, but an error " +
+                "occurred while deleting their associated actions.";
```

Comment:
The message should indicate that the IdP was deleted. But deleting action failed and in stale state.
In both cases stale action ids should present or the other option is not log this in a different file. Given this is an edge case we can improve to that level later and let's make sure data is available to act upon in some way

Reply:
And we can make this log asynchronous